### PR TITLE
[OSX] rename Bitcoin-Qt.app to Bitcoin-Core.app

### PR DESCRIPTION
**Pull request by jonasschnelli**
Bitcoin-Qt is a strange name for a GUI application. IMO we could try to slowly remove the "-Qt" part and try to be consistent with "Bitcoin-Core". The main windows title is already saying that we are running "Bitcoin Core".

### CODE REVIEW

**Comment by laanwj** [comment]
This obviously needs to be done at some point.
However there are some ancillary issues, such as auto-start-on-startup, that need to be considered.
As well as the installer (making sure the old executable, as well as links to it, are deleted on upgrade), as well as making sure QSettings still end up in the same place or are migrated.
These may not be an issue on mac though.

BTW: The eventual DMG in releases is already called `bitcoin-0.10.1-osx.dmg`. No qt in there?

**Comment by jonasschnelli** [comment]
The .dmg name is correct. But when you open the disk image you will find an application called "Bitcoin-Qt".

I'll wait now for other opinions and if there are no reasons to stay with `Bitcoin-Qt`, i think it then makes sense to continue and complete this PR.

**Comment by fanquake** [comment]
Concept ACK

**Comment by jonasschnelli** [comment]
Tested on windows regarding persisted settings (`QSettings`) and autostart. Also tested nsi installer.

If we continue with this, somebody with binary signing capabilities (win/mac) should check how a name change would affect signing.
Process of updating Bitcoin-Qt is also something which should be tested.

**Comment by jonasschnelli** [comment]
A gitian build (with logs to check) can be downloaded here: https://builds.jonasschnelli.ch/pulls/6116/

**Comment by jonasschnelli** [comment]
After discussion this on IRC we came to the point to not changing the actual executable names.
Rolled this back so this PR will only change the OSX Bundle Name (executable and signing entities stays the same).

**Comment by Michagogo** [comment]
And re: the installer, I don't know that that can really be done, considering the installation process for OS X is "drag the .app folder into /Applications". (I'm assuming we don't want to use a .pkg)

**Comment by jonasschnelli** [comment]
"Bitcoin\ Core.app" sounds good to me. I'll change it that way. The installation process will be manually (user need to delete Bitcoin-Qt.app before or after he installed/copied the new Bitcoin-Core.app. I think we don't want to go for a .pkg (uncommon on OSX).

**Comment by jonasschnelli** [comment]
Changing it to "Bitcoin\ Core.app" seems not to be possible. Somehow Automake and co. are not ready to build binaries with names containing whitespaces. I think we should keep it Bitcoin-Core.app (as this PR like to change it to) to avoid heavy-maintainable hacks.

**Comment by Michagogo** [comment]
If it can't be done without a hack, then it's not such a huge deal... It's
just that from the limited experience I have with Macs, having multiple
words separated by something other than spaces is non-standard and I
figured it would look weird, kinda like how the standard installer is a
.dmg with an .app and a shortcut to /Applications. (Also, nit: "Bitcoin\
Core.app" is redundant, since you don't need \ in quotes. (And if I made
that mistake... oops.))

**Comment by laanwj** [comment]
> Changing it to "Bitcoin\ Core.app" seems not to be possible. Somehow Automake and co. are not ready to build binaries with names containing whitespaces.

That's really sad, if true. It's not just a matter of escaping in the right way?

**Comment by jonasschnelli** [comment]
@laanwj:
After trying to use whitespace with escape sequences like '\ ' or taking the whole app name into string quotes and also did a google research on how autotools and whitespace can live together i found out that this is not just a trivial escaping thing.
I try now rename the .app file during the make process before adding it to the disk image. This could be a  quick and stable workaround.

And yes. The travis issue is strange and i can't cure it from my side. Other PRs work fine. I force pushed / rebased this PR some times but the .jar issue is still there.I don't get this.

**Comment by jonasschnelli** [comment]
Added another commit: now `Bitcoin-Core.app` (folder) gets renamed to `Bitcoin Core.app` during the disk image process.
Just started another PR gitian build to test this more authentic: https://builds.jonasschnelli.ch/pulls/6116/

**Comment by jonasschnelli** [comment]
This is not ready yet. Gitian build process bypassed `macdeployqtplus`. It first need a fix for the DMG name as well as for the missing background picture.

**Comment by jonasschnelli** [comment]
Solved. The window title of the disk image name is now "Bitcoin-Core" (instead of Bitcoin-Qt). The app name inside the disk image is now "Bitcoin Core" (with whitespace between "Bitcoin" and "Core"). Tested over gitian.

Ready to ACK and Merge. :)

**Comment by jonasschnelli** [comment]
The travis issues seems to be connected to the JAR comparison tool. I can't see any connection to this PR.

**Comment by jonasschnelli** [comment]
Here is the line where travis fails. Looks like the nativ comparison tool can not be unpacked: https://travis-ci.org/bitcoin/bitcoin/jobs/62393150#L272

**Comment by faizkhan00** [comment]
thanks for noticing that @jonasschnelli , i'm looking into the jar issues today and yea, its buggy all-around (https://github.com/bitcoin/bitcoin/issues/6119)

**Comment by jonasschnelli** [comment]
Travis is now happy with this. Would be nice if we can include this in 0.11.

**Comment by laanwj** [comment]
@theuni Can you take a look here?

**Comment by theuni** [comment]
Sure. I'd like to go through the entire release process with this change on top, just to make sure there's nothing missing. Will do that today.

**Comment by theuni** [comment]
The detached sig create/apply scripts needed a good bit of quoting love to deal with the spaces. My fault there.

I'm fixing those up now.

File: Makefile.am [hunk 1]
```diff
@@ -106,7 +106,7 @@ $(APP_DIST_DIR)/Applications:
 $(APP_DIST_EXTRAS): $(APP_DIST_DIR)/$(OSX_APP)/Contents/MacOS/Bitcoin-Qt
 
 $(OSX_DMG): $(APP_DIST_EXTRAS)
-	$(GENISOIMAGE) -no-cache-inodes -D -l -probe -V "Bitcoin-Qt" -no-pad -r -apple -o $@ dist
+	$(GENISOIMAGE) -no-cache-inodes -D -l -probe -V "Bitcoin-Core" -no-pad -r -apple -o $@ dist
```
Comment (theuni) [inline Makefile.am]: Any reason for not using "Bitcoin Core" here? These set the volume descriptor that OSX ends up using for mounting.

File: contrib/gitian-descriptors/gitian-osx-signer.yml [hunk 2]
```diff
@@ -33,5 +33,5 @@ script: |
 
   tar -xf ${UNSIGNED}
   ./detached-sig-apply.sh ${UNSIGNED} signature.tar.gz
-  ${WRAP_DIR}/genisoimage -no-cache-inodes -D -l -probe -V "Bitcoin-Qt" -no-pad -r -apple -o uncompressed.dmg signed-app
+  ${WRAP_DIR}/genisoimage -no-cache-inodes -D -l -probe -V "Bitcoin-Core" -no-pad -r -apple -o uncompressed.dmg signed-app
```
Comment (theuni) [inline contrib/gitian-descriptors/gitian-osx-signer.yml]: Same here.

**Comment by theuni** [comment]
@jonasschnelli I needed https://github.com/theuni/bitcoin/commit/c2117a6b981bc20048acdff949bef98fd579b45b in order to get signing working. Seems to be working fine. It even hides the old Bitcoin-Qt in Launchpad. Neat.

I'm not 100% sure that this won't cause any issues for upgraders, but I guess we'll find out when we do a real signing for rc1.

**Comment by jonasschnelli** [comment]
@theuni: will pull in you commit as soon as I'm back at the keys.

Upgraders should delete the old version in the /Application folder.
Launchpad uses the bundle identifier - which i kept the same in this PR - to identify the uniqueness.

**Comment by jonasschnelli** [comment]
Pulled in @theuni's change and built again over gitian:
https://builds.jonasschnelli.ch/pulls/6116/

This is how it looks after and before this PR:
bildschirmfoto-2015-05-18-um-15 11 52

File: Makefile.am (hunk 1 above)
Comment (laanwj) [inline Makefile.am]: Agreed @theuni. Let's avoid '-' where possible.

Comment (jonasschnelli) [inline Makefile.am]: Agreed. Fixed.

**Comment by laanwj** [comment]
utACK

**Comment by jonasschnelli** [comment]
It's currently not working:

`/home/ubuntu/build/bitcoin/depends/x86_64-apple-darwin11/native/bin/genisoimage: No such file or directory. Invalid node - 'Core'.`

Now i know why i left the dash there. :)

Will try to find a fix (maybe over renaming) for this.

**Comment by jonasschnelli** [comment]
It looks like `genisoimage`'s `-V` (`volid`) parameter is not whitespace compatible (even with quotes). Therefore i suggest using "Bitcoin-Core" as disk image title.

**Comment by laanwj** [comment]
I suppose the volume name isn't user-visible normally? In any case, not important.

**Comment by jonasschnelli** [comment]
@laanwj: it is visible but not that important. See https://github.com/bitcoin/bitcoin/pull/6116#issuecomment-103054666. It's visible in the left part of a Finder window as well as in the window title of the opened disk-image in Finder.

**Comment by fanquake** [comment]
Post merge ACK, glad this is now done.

**Pull request merged**